### ISSUE TITLE
Allow filtering notifications by Participating/Mentioned

### ISSUE THREAD

**Issue by j-f1** [GitHawkApp/GitHawk#2272]
The filter icon on the top left would have all the options it currently does except for “View All.” There could then be a dropdown similar to the kind that’s shown on issue/PR/repo screens. This dropdown could show the owner/repo if you’ve filtered by one, as well as a choice between “All,” “Unread,” “Participating,” and “Mentioned.” The [API docs](https://developer.github.com/v3/activity/notifications/#notification-reasons) say there’s several _reasons_ notifications are sent. I’d suggest `assign`, `mention`, and `team_mention` for “Mentioned” and `author`, `comment`, and `state_change` for “Participating.”

**Comment by BrianLitwin** [comment]
Brought up something similar in #2065.. as I recall, it's pretty low cost to build and implement. Might have more relevance now w/ push notifications?

**Comment by j-f1** [comment]
> Might have more relevance now w/ push notifications? 

Yes! Would be nice to only get notified for things I’m participating in.

**Comment by rnystrom** [comment]
Open to review some designs/flows/PRs

**Comment by ijm8710** [comment]
Yes!!!! This would be huge. I love how the inbox tab lists summarized activity that I’ve missed in repos I follow but would love to only get notifications for threads I’ve participated in. This and the ongoing work in #2106 are the two things I would most love to see!!

+1
